src/common/main.lua

```diff
@@ -1,28 +1,143 @@
 local json = require(".common.json")
 local utils = require(".common.utils")
 local main = {}
+-- just to ignore lint warnings
+local ao = ao or {}
 
 local camel = utils.camelCase
+
+local function errorHandler(err)
+	return debug.traceback(err)
+end
+
 main.init = function()
-	Owner = Owner or ao.env.Process.Owner
-	Name = Name or "ar.io ANT Registry"
+	ANTS = ANTS or {}
+	-- Example ANT structure
+	-- ANTS["antId"] = {
+	--     Owner = "userId",
+	--     Controllers = {"userId1", "userId2"},
+	-- }
+	USERS = USERS or {}
+	-- Example USERS structure
+	-- USERS["userId"] = {"antId1", "antId2"}
 
 	local ActionMap = {
 		Register = "Register",
 		StateNotice = "State-Notice",
 		AccessControlList = "Access-Control-List",
 	}
 
-	Handlers.add("info", Handlers.utils.hasMatchingTag("Action", "Info"), function(msg)
-		ao.send({
-			Target = msg.From,
-			Name = Name,
-			Owner = Owner,
-			Data = json.encode({
-				Name = Name,
-				Owner = Owner,
-			}),
-		})
+	Handlers.add(camel(ActionMap.Register), Handlers.utils.hasMatchingTag("Action", ActionMap.Register), function(msg)
+		print("Action: " .. ActionMap.Register)
+
+		local antId = msg.Tags["Process-Id"]
+		assert(type(antId) == "string", "Process-Id tag is required")
+
+		if ANTS[antId] then
+			ao.send({
+				Target = msg.From,
+				Action = "Register-Notice-Failure",
+				["Message-Id"] = msg.Id,
+				Data = "ANT is already registered",
+			})
+		else
+			ANTS[antId] = {
+				Owner = nil,
+				Controllers = {},
+			}
+
+			ao.send({
```

Comment:
A wee bit of code commenting here could be helpful for future code readers, e.g.:
```
-- Send a request message for current ANT state to the process. Expect back a State-Notice so that we can update the registered ANT settings.
```